lib/traject/indexer.rb

```diff
@@ -568,15 +568,19 @@ def inspect
 
     def execute(context)
       accumulator = []
-      [@lambda, @block].each do |aProc|
-        next unless aProc
-
-        if aProc.arity == 2
-          aProc.call(context.source_record, accumulator)
+      if @lambda
+        if @lambda.arity == 2
+          @lambda.call(context.source_record, accumulator)
         else
-          aProc.call(context.source_record, accumulator, context)
+          @lambda.call(context.source_record, accumulator, context)
+        end
+      end
+      if @block
+        if @block.arity == 2
+          @block.call(context.source_record, accumulator)
+        else
+          @block.call(context.source_record, accumulator, context)
         end
```

Comment:
Sorry to be adding more comments as I think of them -- if the bottleneck was array creation (which I think barmintor suggested in IRC), can we obtain the same speed benefits by putting the `[@lambda, @block]` array into a class constant like you've done in other places, instead of un-DRY unwrapping of the loop?

It would be nice to keep it DRY if possible -- features we're thinking of, like automatic profiling, might require changes at this point in the code, and keeping it DRY would avoid having to repeat the same changes twice too. (That particular feature might not end up being implemented at this code location, but thinking of it re-triggered my aversion to un-DRY'ing here, heheh). 

The change could be made at the analagous location [here](https://github.com/barmintor/traject/blob/gc-min/lib/traject/indexer.rb#L515) too. 

It would also be even more affirming if the change were benchmarked to provide significant improvements in MRI 2.2, not just 2.1, to further help overcome my aversion to making the code somewhat more confusing, demonstrating it really does provide advantage.